### ISSUE TITLE
Lower layer giving better results 

### ISSUE THREAD

**Issue by Colanim** [hanxiao/bert-as-service#85]
I'm using a custom model, using Bert as service feature vectors as input.
I'm solving the problem of sentence textual similarity. I'm using the SICK dataset, but the STS-B dataset (from GLUE) is similar and could be used as well.

I tried to use the default layer, `-2`, and got a score of **~75%**.

I tried to use concatenation of last layers (as described in the paper), ie. `-1 -2 -3 -4`, but the score didn't improve (actually slightly decreased).

I finally tried a low layer, `-11`, and got a score of **~80%**.

---

**Why a lower layer would give a better score ?**
I don't understand...

**Comment by hanxiao** [comment]
btw below is result from the [original BERT paper](https://arxiv.org/pdf/1810.04805.pdf) (Sect. 5.4) Note that they use pretrained-fixed-BERT to output contextualized word embedding, and then feed to **a randomly initialized two-layer 768-dimensional BiLSTM** before the classification layer. And it is only 0.3 F1 behind fine-tuning the entire model

![image](https://user-images.githubusercontent.com/2041322/49433813-c76e3b00-f7ed-11e8-97e8-e7288272aca9.png)

may also related to #76

**Comment by hanxiao** [comment]
my bad. I forgot to mask the sequence when doing reduce_mean, reduce_max and reduce_mean_max. this is fixed in #94 and now merged into master. please give it a try and may produce different result.

**Comment by hanxiao** [comment]
nice, good to hear. close #85

**Comment by hanxiao** [comment]
FYI, I made a visualization of different BERT layers, may help you understand the `pool_layer` and `pool_strategy`. See: https://github.com/hanxiao/bert-as-service#q-so-which-layer-and-which-pooling-strategy-is-the-best

**Comment by chikubee** [comment]
@hanxiao 
It is stated everywhere that the [CLS] token is the correct sentence representation for a fine tuned model.
I fine tuned bert base uncased model for sentiment analysis task.
The eval accuracy is really good.

I aim to interpret the false positives and what lead to them.
I thought finding similar representations in the training set as the test sample, would give me some insight so as to what lead to the prediction.

I've taken [CLS] token representation from the last layer of the fine tuned model.
The sentences I get similar to a false positive, do not look similar but they do share the same sentiment as predicted.
<img width="1009" alt="Screenshot 2019-08-14 at 11 53 32 AM" src="https://user-images.githubusercontent.com/25073753/62998798-3a09dd00-be8a-11e9-8330-c65da3fa0cf6.png">

What is going on here?
Is it really the correct representation.

I would also like to know how this representation is learnt by CLS token, if all the tokens can look at the left and right context and itself., How is it that CLS is capturing the entire sentence's meaning?

Thanks in advance.
